test/Database/LSMTree/Model/Normal/Session.hs

```diff
@@ -93,23 +97,30 @@ initModel = Model {
     , snapshots = Map.empty
     }
 
+-- | We conservatively model blob reference invalidation: each update after
+-- acquiring a blob reference will invalidate it. We use 'UpdateCounter' to
+-- track updates.
+newtype UpdateCounter = UpdateCounter Word64
+  deriving (Show, Eq, Ord, Num)
+
 data SomeTable where
   SomeTable :: forall k v blob. C k v blob
             => Model.Table k v blob -> SomeTable
 
 instance Show SomeTable where
   show (SomeTable table) = show table
 
+unsafeCoerceTable :: SomeTable -> Model.Table k v blob
+unsafeCoerceTable (SomeTable tbl) = unsafeCoerce tbl
```

Comment:
That's pretty hairy!

Can't we just do some dynamic typing?

Reply:
Using `Dynamic` now